exonum-java-proofs/src/main/java/com/exonum/binding/storage/proofs/map/flat/UncheckedFlatMapProof.java

```diff
@@ -0,0 +1,176 @@
+package com.exonum.binding.storage.proofs.map.flat;
+
+import static com.exonum.binding.hash.Funnels.hashCodeFunnel;
+import static com.exonum.binding.storage.proofs.DbKeyFunnel.dbKeyFunnel;
+
+import com.exonum.binding.hash.HashCode;
+import com.exonum.binding.hash.HashFunction;
+import com.exonum.binding.hash.Hashing;
+import com.exonum.binding.storage.proofs.map.DbKey;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * An unchecked flat map proof, which does not include any intermediate nodes.
+ */
+public class UncheckedFlatMapProof implements UncheckedMapProof {
+
+  private static final HashFunction HASH_FUNCTION = Hashing.defaultHashFunction();
+
+  private final List<MapProofEntry> proofList;
+
+  private final Set<byte[]> requestedKeys;
+
+  UncheckedFlatMapProof(List<MapProofEntry> proofList, Set<byte[]> requestedKeys) {
+    this.proofList = proofList;
+    this.requestedKeys = requestedKeys;
+  }
+
+  @SuppressWarnings("unused") // Native API
+  static UncheckedFlatMapProof fromUnsorted(MapProofEntry[] proofList, byte[][] requestedKeys) {
+    List<MapProofEntry> mapProofEntries = Arrays.asList(proofList);
+    mapProofEntries.sort(Comparator.comparing(MapProofEntry::getDbKey));
+    Set<byte[]> expectedKeys = new HashSet<>(Arrays.asList(requestedKeys));
+    return new UncheckedFlatMapProof(mapProofEntries, expectedKeys);
+  }
+
+  @Override
+  public List<MapProofEntry> getProofList() {
+    return proofList;
+  }
+
+  @Override
+  public CheckedMapProof check() {
```

Comment:
The validator method has to check that the actual proof proves that the requested keys are either in the map or not in the map.